exchanges/ftx/ftx_websocket_test.go

```diff
@@ -0,0 +1,209 @@
+package ftx
+
+import (
+	"testing"
+	"time"
+
+	"github.com/thrasher-corp/gocryptotrader/currency"
+	exchange "github.com/thrasher-corp/gocryptotrader/exchanges"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/asset"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/order"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/stream"
+)
+
+func parseRaw(t *testing.T, input string) interface{} {
+	t.Helper()
+	pairs := currency.Pairs{
+		currency.Pair{
+			Base:  currency.NewCode("BTC"),
+			Quote: currency.NewCode("USDT"),
+		},
+	}
+	f := FTX{
+		exchange.Base{
+			CurrencyPairs: currency.PairsManager{
+				Pairs: map[asset.Item]*currency.PairStore{
+					asset.Spot: {
+						Available: pairs,
+						Enabled:   pairs,
+						ConfigFormat: &currency.PairFormat{
+							Delimiter: "^",
+							Uppercase: true,
+						},
+					},
+				},
+				// RequestFormat: &currency.PairFormat{
+				// 	Delimiter: ">",
+				// 	Uppercase: false,
+				// },
+				// UseGlobalFormat: true,
+			},
+			Websocket: &stream.Websocket{
+				DataHandler: make(chan interface{}),
+			},
+		},
+	}
+	go func() {
+		f.wsHandleData([]byte(input))
+	}()
+	// Give the channel limited time to yield the data.
+	ticker := time.NewTicker(100 * time.Second)
```

Comment:
Given the < 2 second execution time for me, I'd much prefer a more reasonable deadline to be defined in the event of a failure. <= 10 second maximum please